### PR TITLE
storageclass & volume docs

### PR DESCRIPTION
**Pull request by min-zh**
Signed-off-by: zhangmin <arminzhang@yunify.com>

### CODE REVIEW

**Review by stoneshi-yunify — APPROVED** [review]
/lgtm

File: content/en/docs/project-user-guide/storage/volume-snapshots.md [hunk 1]
```diff
@@ -8,9 +8,24 @@ weight: 2130
 ---
 
 ## Introduction
+Many storage systems provide the ability to create a "snapshot" of a persistent volume. 
+A snapshot represents a point-in-time copy of a volume. 
+A snapshot can be used either to provision a new volume (pre-populated with the snapshot data) 
+or to restore the existing volume to a previous state (represented by the snapshot).
+
+On KubeSphere, conditions for Volume Snapshot are:
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volume-snapshots.md:16]: conditions -> requirements

File: content/en/docs/project-user-guide/storage/volume-snapshots.md [hunk 2]
```diff
@@ -8,9 +8,24 @@ weight: 2130
...
+- With 1.17+ Kubernetes
+- Underlying storage plugin support Snapshot
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volume-snapshots.md:18]: support -> supports

File: content/en/docs/project-user-guide/storage/volume-snapshots.md [hunk 3]
```diff
@@ -8,9 +8,24 @@ weight: 2130
...
+A snapshot represents a point-in-time copy of a volume. 
+A snapshot can be used either to provision a new volume (pre-populated with the snapshot data) 
+or to restore the existing volume to a previous state (represented by the snapshot).
+
+On KubeSphere, conditions for Volume Snapshot are:
+- With 1.17+ Kubernetes
+- Underlying storage plugin support Snapshot 
 
 ## Create Volume Snapshot
+Volume Snapshot could be created from an exist volume on the volume detail page.
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volume-snapshots.md:21]: exist -> existing

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 4]
```diff
@@ -33,7 +33,7 @@ KubeSphere can get supported features of underlying storage plugin that called `
 The console display only supported features in `Volume Detail Page`.
 More about `Storage Capability` see [design](https://github.com/kubesphere/community/blob/master/sig-storage/concepts-and-designs/storage-capability-interface.md)
 
-![Volume Feature](/images/storage/volume_features.png)
+![Volume Feature](/images/storage/volume-features.png)
 
 **Node**: Some in-tree or special CSI plugins may not covered by **Storage Capability**.
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volumes.md:38]: may not -> may not be

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 5]
```diff
@@ -33,7 +33,7 @@ KubeSphere can get supported features of underlying storage plugin that called `
 The console display only supported features in `Volume Detail Page`.
 More about `Storage Capability` see [design](https://github.com/kubesphere/community/blob/master/sig-storage/concepts-and-designs/storage-capability-interface.md)
 
-![Volume Feature](/images/storage/volume_features.png)
+![Volume Feature](/images/storage/volume-features.png)
 
 **Node**: Some in-tree or special CSI plugins may not covered by **Storage Capability**. 
 If Kubesphere not display the right features in your cluster, you could adjust according to [method](https://github.com/kubesphere/kubesphere/issues/2986).
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volumes.md:39]: not display -> did not display

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 6]
```diff
@@ -33,7 +33,7 @@ KubeSphere can get supported features of underlying storage plugin that called `
 The console display only supported features in `Volume Detail Page`.
```
Comment (stoneshi-yunify) [inline content/en/docs/project-user-guide/storage/volumes.md:33]: supported -> supports

File: content/en/docs/project-user-guide/storage/volume-snapshots.md (hunk 1 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volume-snapshots.md:16]: OK

File: content/en/docs/project-user-guide/storage/volume-snapshots.md (hunk 2 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volume-snapshots.md:18]: OK

File: content/en/docs/project-user-guide/storage/volume-snapshots.md (hunk 3 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volume-snapshots.md:21]: ok

File: content/en/docs/project-user-guide/storage/volumes.md (hunk 4 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volumes.md:38]: ok

File: content/en/docs/project-user-guide/storage/volumes.md (hunk 5 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volumes.md:39]: ok

File: content/en/docs/project-user-guide/storage/volumes.md (hunk 6 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volumes.md:33]: I mean `passive`

**Review by stoneshi-yunify — APPROVED** [review]
/lgtm

File: content/en/docs/cluster-administration/storageclass.md [hunk 7]
```diff
@@ -1,8 +1,41 @@
...
+
+- Storage System: As **hyperkube** images were [deprecated since 1.17](https://github.com/kubernetes/kubernetes/pull/85094), in-tree Ceph RBD may not work without **hyperkube**.
+Fortunately, [rbd provisioner](https://github.com/kubernetes-incubator/external-storage/tree/master/ceph/rbd) could be used as a substitute, 
+which format is the same as in-tree Ceph RBD. The only different parameter is the **provisioner** here is the **storage system**. 
+So if you has installed and choose to use rbd provisioner, the parameter **Storage System** should be **ceph.com/rbd**.
+If you use in-tree Ceph RBD, the parameter **storage system** should be **kubernetes.io/rbd**. 
+
+### Glusterfs
+Glusterfs is also an in-tree storage plugin on Kubernetes. 
+StorageClass parameters could be seen in [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes/#glusterfs).
```
Comment (Sherlock113) [inline content/en/docs/cluster-administration/storageclass.md:30]: StorageClass parameters could be seen in [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes/#glusterfs).
->
For more information about StorageClass parameters, see [Glusterfs in Kubernetes Documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/#glusterfs).

File: content/en/docs/cluster-administration/storageclass.md [hunk 8]
```diff
@@ -1,8 +1,41 @@
...
+
+### QingCloud CSI
+QingCloud CSI is the storage plugin on Kubernetes for the disk of QingCloud. 
+Before creating StorageClass, QingCloud CSI plugin should be installed on the cluster.
```
Comment (Sherlock113) [inline content/en/docs/cluster-administration/storageclass.md:34]: ->
Before you create the StorageClass, QingCloud CSI plugin should be installed on the cluster.

File: content/en/docs/cluster-administration/storageclass.md [hunk 9]
```diff
@@ -1,8 +1,41 @@
...
+StorageClass parameters could be seen in [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes/#glusterfs).
+
+### QingCloud CSI
+QingCloud CSI is the storage plugin on Kubernetes for the disk of QingCloud. 
+Before creating StorageClass, QingCloud CSI plugin should be installed on the cluster. 
+See [QingCloud-CSI Installation](https://github.com/yunify/qingcloud-csi#installation) for more information about installation.
+The StorageClass Parameters could be seen in [user guide](https://github.com/yunify/qingcloud-csi/blob/master/docs/user-guide.md#set-storage-class).
+
+### Custom StorageClass
+If none of the above storage plugins used, you could choose to create custom StorageClass for other plugins.
```
Comment (Sherlock113) [inline content/en/docs/cluster-administration/storageclass.md:39]: If none of the above storage plugins are used, you could choose to create custom Storage Classes for other plugins.

File: content/en/docs/cluster-administration/storageclass.md [hunk 10]
```diff
@@ -1,8 +1,41 @@
...
+### QingCloud CSI
+QingCloud CSI is the storage plugin on Kubernetes for the disk of QingCloud. 
+Before creating StorageClass, QingCloud CSI plugin should be installed on the cluster. 
+See [QingCloud-CSI Installation](https://github.com/yunify/qingcloud-csi#installation) for more information about installation.
+The StorageClass Parameters could be seen in [user guide](https://github.com/yunify/qingcloud-csi/blob/master/docs/user-guide.md#set-storage-class).
+
+### Custom StorageClass
+If none of the above storage plugins used, you could choose to create custom StorageClass for other plugins. 
+**Storage System** represents the parameter [provisioner](https://kubernetes.io/docs/concepts/storage/storage-classes/#provisioner).
+More information about StorageClass, see [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes).
```
Comment (Sherlock113) [inline content/en/docs/cluster-administration/storageclass.md:41]: More information about StorageClass, see [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes).
->
For more information about Storage Classes, see [Kubernetes Documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/).

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 11]
```diff
@@ -1,10 +1,46 @@
...
+![Volume Feature](/images/storage/volume-features.png)
+
+**Node**: Some in-tree or special CSI plugins may not be covered by **Storage Capability**. 
+If Kubesphere did not display the right features in your cluster, you could adjust according to [method](https://github.com/kubesphere/kubesphere/issues/2986).
+
+## Volume Monitoring
+KubeSphere get metric data of PVC with FileSystem mode from Kubelet to monitor volumes including capacity usage and inode usage. 
+![Monitoring](/images/storage/volume-monitoring.png)
+
+Illustration can be referred to [Research on Volume Monitoring](https://github.com/kubesphere/kubesphere/issues/2921).
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/storage/volumes.md:45]: Illustration can be referred to [Research on Volume Monitoring](https://github.com/kubesphere/kubesphere/issues/2921).
->
For more information about Volume Monitoring illustrations, see [Research on Volume Monitoring](https://github.com/kubesphere/kubesphere/issues/2921).

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 12]
```diff
@@ -1,10 +1,46 @@
...
+volumes cloud be attached on containers' path.
+![Attach](/images/storage/attach-volume.png)
+
+## Volume Features
+Volume Features include: 
+- Clone Volume
+- Create Volume Snapshot
+- Expand Volume 
+
+KubeSphere can get supported features of underlying storage plugin that called `Storage Capability`.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/storage/volumes.md:32]: Just to confirm here, does "called" here mean "调用" in Chinese? If so, please change this sentence to:

KubeSphere can get supported features of the underlying storage plugin that calls `Storage Capability`.

If you mean the name, change the sentence to:

KubeSphere can get supported features of the underlying storage plugin called `Storage Capability`.

File: content/en/docs/project-user-guide/storage/volumes.md (hunk 6 above)
Comment (Sherlock113) [reply content/en/docs/project-user-guide/storage/volumes.md:33]: ->
The console only displays supported features in `Volume Detail Page`.

I think the word "display" is used as a "verb" here and that "supported" is used as an adjective.

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 13]
```diff
@@ -1,10 +1,46 @@
...
+The console display only supported features in `Volume Detail Page`.
+More about `Storage Capability` see [design](https://github.com/kubesphere/community/blob/master/sig-storage/concepts-and-designs/storage-capability-interface.md)
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/storage/volumes.md:34]: More about `Storage Capability` see [design](https://github.com/kubesphere/community/blob/master/sig-storage/concepts-and-designs/storage-capability-interface.md)
->
For more information about `Storage Capability`, see [Design](https://github.com/kubesphere/community/blob/master/sig-storage/concepts-and-designs/storage-capability-interface.md#Design).

File: content/en/docs/project-user-guide/storage/volumes.md [hunk 14]
```diff
@@ -1,10 +1,46 @@
...
+
+![Volume Feature](/images/storage/volume-features.png)
+
+**Node**: Some in-tree or special CSI plugins may not be covered by **Storage Capability**. 
+If Kubesphere did not display the right features in your cluster, you could adjust according to [method](https://github.com/kubesphere/kubesphere/issues/2986).
+
+## Volume Monitoring
+KubeSphere get metric data of PVC with FileSystem mode from Kubelet to monitor volumes including capacity usage and inode usage.
```
Comment (Sherlock113) [inline content/en/docs/project-user-guide/storage/volumes.md:42]: KubeSphere get metric data
->
KubeSphere gets metric data

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments. Thanks.

File: content/en/docs/cluster-administration/storageclass.md [hunk 15]
```diff
@@ -1,8 +1,41 @@
...
+Glusterfs is also an in-tree storage plugin on Kubernetes. 
+StorageClass parameters could be seen in [Kubernetes Docs](https://kubernetes.io/docs/concepts/storage/storage-classes/#glusterfs).
+
+### QingCloud CSI
+QingCloud CSI is the storage plugin on Kubernetes for the disk of QingCloud. 
+Before creating StorageClass, QingCloud CSI plugin should be installed on the cluster. 
+See [QingCloud-CSI Installation](https://github.com/yunify/qingcloud-csi#installation) for more information about installation.
+The StorageClass Parameters could be seen in [user guide](https://github.com/yunify/qingcloud-csi/blob/master/docs/user-guide.md#set-storage-class).
+
+### Custom StorageClass
```
Comment (FeynmanZhou) [inline content/en/docs/cluster-administration/storageclass.md:38]: @min @stoneshi-yunify How to add NFS as a new StorageClass in KubeSphere StorageClass configuration page?
I don't find any description from the [K8s document](https://kubernetes.io/docs/concepts/storage/storage-classes/).

Comment (min-zh) [reply content/en/docs/cluster-administration/storageclass.md:38]: NFS doesn't provide an internal provisioner, but an external provisioner can be used. We use [nfs-client](https://github.com/kubernetes-retired/external-storage/tree/master/nfs-client) that is an external provisioner for NFS.

File: content/en/docs/project-user-guide/storage/volumes.md (hunk 12 above)
Comment (min-zh) [reply content/en/docs/project-user-guide/storage/volumes.md:32]: I mean the name.

**Comment by Sherlock113** [comment]
/lgtm
Please merge this pr if you do not have other questions. Thanks. @FeynmanZhou

File: content/en/docs/cluster-administration/storageclass.md [hunk 16]
```diff
@@ -1,8 +1,145 @@
...
+| userSecretName | The name of Ceph Secret for userId to map RBD image | 
+| userSecretNamespace | The namespace for *userSecretName* | 
+| fsType | fsType that is supported by kubernetes | 
+| imageFormat | Ceph RBD image format, "1" or "2" | 
+| imageFeatures| This parameter is optional and should only be used if you set imageFormat to “2”| 
+
+For more information about StorageClass parameters, see [Ceph RBD in Kubernetes Documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/#ceph-rbd).
+
+### Custom StorageClass
+If none of the above storage plugins used, custom storage classes for other plugins could be chose.
```
Comment (Sherlock113) [inline content/en/docs/cluster-administration/storageclass.md:121]: ->
In addition to the above storage plugins, you can also create custom storage classes of other plugins.

**Review by Sherlock113 — COMMENTED** [review]
Please have a look at my comments. Besides, please global search "filed" in this pr and replace it with "field". I think you mean "字段" in Chinese.

This tutorial looks much clearer after the modification. Thanks.

File: content/en/docs/cluster-administration/storageclass.md [hunk 17]
```diff
@@ -1,8 +1,145 @@
...
+| Supported Access Mode | Annotation `metadata.annotations[storageclass.kubesphere.io/supported-access-modes]`.It tells KubeSphere which [`Access Mode`](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes) supported| 
+
+Other settings are always customized for the special plugin which are always defined in the filed of [`parameter`](https://kubernetes.io/docs/concepts/storage/storage-classes/#parameters) in storage class.
+They will be described in the detailed section of volume plugin.
+
+### QingCloud CSI
+QingCloud CSI is a CSI plugin on Kubernetes for the disk of QingCloud. Storage classes of QingCloud CSI could be created on KubeSphere console.
+
+#### Prerequisite
+- KubeSphere has been installed on QingCloud either public or private so that you can use cloud disk.
```
[the remark this answers is not in the document]
Comment (min-zh) [reply content/en/docs/cluster-administration/storageclass.md:52]: OK

**Comment by min-zh** [comment]
> Please have a look at my comments. Besides, please global search "filed" in this pr and replace it with "field". I think you mean "字段" in Chinese.
> 
> This tutorial looks much clearer after the modification. Thanks.

Thank you for review. Write ability improves quickly with you and pengfei's  help.

**Comment by Sherlock113** [comment]
@FeynmanZhou Please have a look at this. Thanks.

**Comment by FeynmanZhou** [comment]
> @FeynmanZhou Please have a look at this. Thanks.

ok, it looks better than before. I will test it accordingly.

**Comment by stoneshi-yunify** [comment]
@FeynmanZhou @Sherlock113 can you approve and merge this PR? Then I will modify with my own branch based on that.

**Comment by stoneshi-yunify** [comment]
@Sherlock113 @FeynmanZhou I have modified some contents according to last meeting, please review it again. thanks.

**Comment by Sherlock113** [comment]
@FeynmanZhou Please merge this and I will update the wording.

**Review by rayzhou2017 — APPROVED** [review]
/lgtm

**Pull request merged**